liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,500 @@
+          ).dividedBy(data.tokens[token.address].exchangeRateMult);
+
+          ts(
+            `User ${[account.address]
+            } supplied ${Math.round(token.supply_balance_underlying.value)
+            } ${data.tokens[token.address].symbol
+            } ( ${Math.round(data.supply[token.address][account.address])
+            } cTokens ) `,
+          );
+        }
+        /* eslint-enable eqeqeq */
+      }); // end token loop
+      /* eslint-enable no-param-reassign */
+    }); // end account loop
+    // #endregion
+  };
+}
+
+function provideHandleTransaction(data) {
+  return async function handleTransaction(txEvent) {
+    // #region Filter logs and look for new account activity.
+    // Filter all new transactions and look for new accounts to track.
+    const borrowString = 'event Borrow(address borrower, uint256 borrowAmount, uint256 accountBorrows, uint256 totalBorrows)';
+    const exitMarketString = 'event MarketExited(address cToken, address account)';
+
+    const exitMarketEvents = txEvent.filterLog(exitMarketString);
+    exitMarketEvents.forEach((exitEvent) => {
+      ts(`${exitEvent.args.account} exited from ${exitEvent.args.account} `);
+      data.newAccounts.push(exitEvent.args.account);
+    });
+    const borrowEvents = txEvent.filterLog(borrowString);
+    borrowEvents.forEach((borrowEvent) => {
+      ts(`${borrowEvent.args.borrower} borrowed ${borrowEvent.args.borrowAmount} `);
+      data.newAccounts.push(borrowEvent.args.borrower);
+    });
+    // #endregion
+
+    // Return zero findings
+    return [];
+  };
+}
+
+function provideHandleBlock(data) {
+  return async function handleBlock(blockEvent) {
+    ts(`Starting block ${blockEvent.blockNumber}`);
+    const findings = [];
+    const { comptrollerContract, oneInchContract } = data;
+
+    // #region Add new Accounts
+    ts(
+      `Currently tracking ${Object.keys(data.accounts).length
+      } accounts and adding ${data.newAccounts.length
+      } account(s) this block. ${data.totalNewAccounts
+      } total accounts added since start.`,
+    );
+    data.newAccounts.forEach((newAccount) => {
+      const account = newAccount.toLowerCase();
+      // Initialize account. New accounts will get updated in the block section
+      /* eslint-disable no-param-reassign */
+      data.accounts[account] = {};
+      /* eslint-enable no-param-reassign */
+    });
+    /* eslint-disable no-param-reassign */
+    data.totalNewAccounts += data.newAccounts.length;
+    data.newAccounts = [];
+    /* eslint-enable no-param-reassign */
+    // #endregion
+
+    // #region Update Balances on zero health accounts
+    const filteredAccounts = [];
+    Object.keys(data.accounts).forEach((currentAccount) => {
+      if (data.accounts[currentAccount].health == null
+        || data.accounts[currentAccount].health === 0) {
+        filteredAccounts.push(currentAccount);
+        // Zero account balances
+        /* eslint-disable no-param-reassign */
```

Comment:
Actually, given how often this eslint setting is toggle on/off, perhaps the setting should just be left disabled for the entire function.

Reply:
Agreed. Actually the destructure will help in this regard

Reply:
Less toggling of linting on commit da9b6aa2a0142c6d6330c6f3e78b4d103959dad8